lib/srv/desktop/audit.go

```diff
@@ -145,6 +181,47 @@ func (s *WindowsService) onClipboardReceive(ctx context.Context, emitter events.
 	s.emit(ctx, emitter, event)
 }
 
+// onSharedDirectoryAnnounce does not emit an audit event, but rather registers a directory's
```

Comment:
So we're only emitting the audit event after the desktop acknowledges the share?

Do you think it would be better to emit the event when the user initiates the share instead? This way we would catch all user attempts in the audit log, even if they fail for some reason.

Reply:
The issue with emitting it when the user initiates the share is that we don't have the information to fill out the `Status` field (currently the `Succeeded` field, but the plan is to change it). I tried to formulate an argument for it in the section in the description above

> One point of note is that as currently designed, its technically possible that a SharedDirectoryAnnounce event is sent, and the server never responds, thus no audit event ever shows up. We could ameliorate this by adding a separate DesktopSharedDirectoryAnnounce (DesktopSharedDirectoryAttempt?) event which is emitted upon that initial announcement.
>
> I don't think such an event is necessary, because it doesn't represent shared data between the machines and therefore I wouldn't predict it to be a security concern. The DesktopSharedDirectoryStart event is itself primarily an aesthetic touch to give the audit logs a coherent narrative (it itself isn't representative shared data between machines). It would be aesthetically unappealing for every successful directory sharing connection to have two audit events.

Granted I can see the argument for the start event being more than an aesthetic touch -- for example, if somebody were attempting to hack the directory sharing system, then it would be relevant that an attempt was made even if that attempt was rejected.

Another option besides the one I suggested (2 events on start -- DesktopSharedDirectoryAttempt + DesktopSharedDirectoryStart) is to cache the `SharedDirectoryAnnounce`, then start a 1 second timer waiting for a `SharedDirectoryAcknowledge`, and if we don't receive one, consider that Attempt a failure (in which case we can use just one event for the whole process, with a rare edge case wherein a slow connection sees a failure followed by a success).

Reply:
Thinking about it further, a premise in my quoted argument above is that the only security relevant information is that data was transferred between the machines. Taking that to its logical conclusion, the only events we need to log are reads and writes, and we can leave out this event as well as some of the others currently suggested in the rfd PR:

- ~`Shared Directory Announce/Acknowledge` --> "desktop.directory.start"~
- ~`Shared Directory Create` --> "desktop.directory.create"~
- `Shared Directory Read` --> "desktop.directory.read"
- `Shared Directory Write` --> "desktop.directory.write"
- ~`Shared Directory Delete` --> "desktop.directory.delete"~
- ~`Shared Directory Move` --> "desktop.directory.move"~